src/main/kotlin/uk/gov/justice/digital/hmpps/prisonercontactregistry/service/PrisonerContactRegistryService.kt

```diff
@@ -45,6 +55,44 @@ class PrisonerContactRegistryService(private val prisonApiClient: PrisonApiClien
     return contacts
   }
 
+  @Throws(VisitorNotFoundException::class, DateRangeNotFoundException::class)
+  fun getBannedDateRangeForPrisonerContacts(
+    prisonerId: String,
+    visitorIds: List<Long>,
+    fromDate: LocalDate,
+    toDate: LocalDate,
+  ): DateRangeDto {
+    val dateRange = DateRangeDto(fromDate, toDate)
+
+    val contacts = getContactById(prisonerId)
+    val visitors = contacts.filter { visitorIds.contains(it.personId) }
+    if (visitors.size != visitorIds.size) {
+      throw VisitorNotFoundException(message = "Not all visitors provided ($visitorIds) are listed contacts for prisoner $prisonerId")
+    }
+
+    val visitorBanRestrictions = visitors
+      .flatMap { it.restrictions }
+      .filter { it.restrictionType == "BAN" }
+
```

Comment:
your could if you want to put : 

```
    val contacts = getContactById(prisonerId)
    val visitors = contacts.filter { visitorIds.contains(it.personId) }
    if (visitors.size != visitorIds.size) {
      throw VisitorNotFoundException(message = "Not all visitors provided ($visitorIds) are listed contacts for prisoner $prisonerId")
    }

    val visitorBanRestrictions = visitors
      .flatMap { it.restrictions }
      .filter { it.restrictionType == "BAN" }
```
In a function :
`
val visitorBanRestrictions = getBannedRestrictions(prisonerId, visitorIds)`